contrib/inventory/gce_googleapiclient.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+"""Usage provided by docoptcfg using the global DOCOPT_USAGE constant"""
+
+from __future__ import print_function
+
+from os.path import basename
+from sys import argv
+
+import logging as log
+import json
+
+from docoptcfg import docoptcfg
+from docoptcfg import DocoptcfgFileError
+
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+
+ENV_PREFIX = 'GCE_'
+
+DOCOPT_USAGE = """
+Google Cloud Engine Dynamic Inventory
+=====================================
+
+Before using:
+
+- Authentication: this script uses the same authentication as gcloud command line. So, set it up before.
+- Dependencies: it depends mainly on google-api-python-client and docoptcfg. So:
+
+$ pip install google-api-python-client docoptcfg
+
+Usage:
+    {0} [options]
+
+Options:
+    -a API_VERSION --api-version=API_VERSION    The API version used to connect to GCE [default: v1]
+    -c CONFIG_FILE --config=CONFIG_FILE         Path to the config file (see docoptcfg docs) [default: ./gce_googleapiclient.ini]
+    -l --list                                   List all hosts (needed by Ansible, but actually doesn't do anything)
+    -p PROJECT --project=PROJECT                The GCE project where you want to get the inventory
+    -z ZONE --zone=ZONE                         The GCE zone where you ant to get the inventory
+
+All the parameters can also be set as environment variables using the 'GCE_' prefix (i.e. {1}API_VERSION=beta).
+""".format(basename(argv[0]), ENV_PREFIX)
+
+def list_all_projects(api_version='v1'):
+    project_ids = []
+
+    credentials = GoogleCredentials.get_application_default()
+
+    service = discovery.build('cloudresourcemanager', api_version, credentials=credentials)
+    # pylint: disable=no-member
+    request = service.projects().list()
+
+    while request is not None:
+        response = request.execute()
+
+        for project in response['projects']:
+            project_ids.append(project['projectId'])
+
+        # pylint: disable=no-member
+        request = service.projects().list_next(previous_request=request,
+                                               previous_response=response)
+
+    return project_ids
+
+def list_instances(project_id, zone, api_version='v1'):
```

Comment:
Maybe we should log the project_id passed to the script as debug. We could enable debug with --debug or GCE_ANSIBLE_INV_DEBUG